plugins/gtkui/actionhandlers.c

```diff
@@ -130,25 +130,75 @@ set_file_filter (GtkWidget *dlg, const char *name) {
     return flt;
 }
 
+static GtkFileFilter *
+set_file_filter_win32 (GtkWidget *dlg, const char *name) {
+    if (!name) {
+        name = _("Supported sound formats");
+    }
+    static char extlist[10000];
+
+    // Build extension list
+    if (extlist[0] == '\0') {
+        DB_decoder_t **codecs = deadbeef->plug_get_decoder_list ();
+        for (int i = 0; codecs[i]; i++) {
+            if (codecs[i]->exts && codecs[i]->insert) {
+                const char **exts = codecs[i]->exts;
+                for (int e = 0; exts[e]; e++) {
+                    char buf[100];
+                    snprintf (buf, sizeof (buf), "*.%s;", exts[e]);
+                    strcat (extlist, buf);
+                }
+            }
+        }
+    }
+
+    GtkFileFilter* flt;
+
+    flt = gtk_file_filter_new ();
+    gtk_file_filter_set_name (flt, name);
+    gtk_file_filter_add_pattern (flt, extlist);
+    gtk_file_chooser_add_filter (GTK_FILE_CHOOSER (dlg), flt);
+    gtk_file_chooser_set_filter (GTK_FILE_CHOOSER (dlg), flt);
+
+    flt = gtk_file_filter_new ();
+    gtk_file_filter_set_name (flt, _("All files (*)"));
+    gtk_file_filter_add_pattern (flt, "*");
+    gtk_file_chooser_add_filter (GTK_FILE_CHOOSER (dlg), flt);
+    return flt;
+}
+
 gboolean
 action_open_files_handler_cb (void *userdata) {
-    GtkWidget *dlg = gtk_file_chooser_dialog_new (_("Open file(s)..."), GTK_WINDOW (mainwin), GTK_FILE_CHOOSER_ACTION_OPEN, GTK_STOCK_CANCEL, GTK_RESPONSE_CANCEL, GTK_STOCK_OPEN, GTK_RESPONSE_OK, NULL);
+#if defined _WIN32 && GTK_CHECK_VERSION(3,20,0)
+    GtkWidget *dlg = gtk_file_chooser_native_new (_("Open file(s)..."), GTK_WINDOW (mainwin), GTK_FILE_CHOOSER_ACTION_OPEN, NULL, NULL);
+#else
+    GtkWidget *dlg = gtk_file_chooser_dialog_new (_("Open file(s)..."), GTK_WINDOW (mainwin), GTK_FILE_CHOOSER_ACTION_OPEN, GTK_STOCK_CANCEL, GTK_RESPONSE_CANCEL, GTK_STOCK_OPEN, GTK_RESPONSE_ACCEPT, NULL);
+#endif
 
+#ifdef _WIN32
+    set_file_filter_win32 (dlg, NULL);
```

Comment:
as I understand gtk native filechooser requirest different filtering, or?
I'd suggest to merge this #ifdef block with the one over it, so that GTK2 version still uses normal set_file_filter